### ISSUE TITLE
jemoji plugin interferes with Polyglot leading space in href

### ISSUE THREAD

**Issue by bertolinocastro** [untra/polyglot#120]
Hi! While writing a post that contains a Github Flavored emoji like :grin:, all my links which start with the leading space, that Polyglot requires so it does not parse them, simply get parsed.

Based on my tests, it looks like that the jemoji plugin is parsing the entire file and removing the leading blank spaces or Polyglot has some undefined behaviour when there is a simple addition of an image by jemoji.

Is there some other way to prevent href from being parsed by Polyglot?

I really can't allow the link to be parsed, as it's the language switcher...

**Comment by untra** [comment]
so I will admit that I regret my decision to use use leading spaces as a secret marker to avoid a regex matching a url.
that is outright a bad design, in particular with how it affects urls that users don't want to relativized.

It appears this plugin [jemoji](https://github.com/jekyll/jemoji) along the [jekyll spaceship](https://github.com/untra/polyglot/issues/116#issuecomment-668349136) plugin both preprocess pages with a regex like jekyll-polyglot does, and wrangles the urls I recommend wrangling to prevent processing.

This is bad design, and thats my bad. **I would like some advice on how to address this, ideally with defining relative `href="..."` urls that shouldn't be relativized by polyglot.**

> Is there some other way to prevent href from being parsed by Polyglot?
there should be, I don't know how to provide it given these circumstances 😬 any suggestions?

-sam

**Comment by bertolinocastro** [comment]
Well, I'm quite new to using Jekyll made pages, so I don't know what kind of tricks are possible to be made by a plugin in the jekyll pipeline. But the first things I could think right now were these:

1. some kind of Liquid comment or tag/marker that Polyglot should check whether the href lies in between or not. This would be something that has an starting and ending comment/tag, like:  
    ```markdown
    {% comment %} START_POLYGLOT_COMMENT {% endcomment %}
    <a href="...">link</a>
    [link]( /example.com)
    {% comment %} END_POLYGLOT_COMMENT {% endcomment %}
    ```

2. Using a liquid filter:  
    ```markdown
    {% polyglot_starting_liquid_filter %}
    <a href="...">link</a>
    [link]( /example.com)
    {% polyglot_ending_liquid_filter %}
    ```
    In this case, it could have some sort of hidden variable being defined by the filters, so the Polyglot ruby code would check if it's true or not before trying to relativizing it. (it it's applicable in ruby language.)

3. or in case of specifically a html <a> tag, a "wildcard" attribute:  
    ```markdown
    <a somepolyglot-attribute-here="" href="">link</a>
    ```

Honestly speaking, I really think the third one is a bad strategy. Also, the first and third ones would make Polyglot regex steps do some new processing before and after the matched href, and therefore increase the building time.

IMO, the 2nd one would be the cleanest solution, but, again, I don't know if it's possible with the Jekyll means.

**Comment by GaLaKtIkUs** [comment]
Hi,
Any update regarding this issue? I'm facing the same situation.

**Comment by untra** [comment]
Hi @GaLaKtIkUs 👋 ,

Sorry I have not worked on this.
Probably making a liquid filter to address this is the right move. That seems straightforward. I'll try and find some time to knock this out.

**Comment by untra** [comment]
alrighty @bertolinocastro and @GaLaKtIkUs ! I just released [jekyll-polyglot-1.4.0](https://rubygems.org/gems/jekyll-polyglot/versions/1.4.0) , which now has a new liquid tag `{% static_href %}`  [for disabling url relativization](https://github.com/untra/polyglot#disabling-url-relativizing). Let me know if this helps with url generation and the jemoji plugin.

hopefully this is a better solution than the leading space in the `href`.
Cheers!

**Comment by bertolinocastro** [comment]
Thank you, @untra ! I'm sure it solved the problem in a nice manner. I'm gonna test it soon and if any problems occur, I get in touch here again.
